llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp

```diff
@@ -909,6 +909,10 @@ class AMDGPUOperand : public MCParsedAsmOperand {
   bool isWaitVDST() const;
   bool isWaitEXP() const;
 
+  auto getPredicate(std::function<bool(const AMDGPUOperand &Op)> P) const {
```

Comment:
Wouldn't it be simpler to implement a couple more named predicate methods in `AMDGPUOperand`, e.g.:
```
  bool isGLC() { return getImm() & CPol::GLC; }
```
Anyway I don't object if you want to do it this way, but it took me a while to understand.

Reply:
I think I'd prefer to have a single generic helper and use that to avoid scattering details of operand definitions over multiple places. Ultimately, the idea is to have an interface to AsmParser that would allow implementing a new kind of operands with a couple lines in a single place in an intuitive manner and not being wrong thinking it does the expected thing.

Reply:
OK. In the long term do you think we should aim for a deisgn where all operand predicate strings in .td files are expressions involving `Op` (like `Op.isCPol()`) instead of just names of predicate methods (like `isCPol`)?

Reply:
I don't know. I see it more an implementation detail and would just do what seems most natural given the current state of things. My immediate impression is that `getPredicate()` basically works around AsmMatcherEmitter expecting the predicate be an operand method, which doesn't seem to work well for non-trivial cases.

Speaking more generally, I would try to design it in such a way that wouldn't require specifying any predicate names explicitly at all, whether it's with or without the object expression.